esmvalcore/_config/_config.py

```diff
@@ -37,6 +37,7 @@ def _get_project_mappings(project):
     config = {}
     config_paths = [
         importlib_files("esmvalcore._config"),
+        Path.home() / ".esmvaltool",
```

Comment:
As far as I understand it, it is quite uncommon and unintended that users change the `config-developer.yml` file. They do change the `config-user.yml` file that lives in `~/.esmvaltool` and it might be an idea to add a config option there. But then again, if you are willing to change something in `~/.esmvaltool`, why can this not be done in the sub-directory?

Reply:
The config-user file is an argument of the esmvaltool command (and defaults to one in ~/.esmvaltool), and the config-dev file is specified in the config-user file 

If we agree that a model developer user should be allowed to change what the model does output, and so that he should be able to change the mapping dicts, then he has to be allowed to run two instances of ESMValTool at the same time, one with a legacy mapping, and another one with a changed mapping. So, yes, he should at least have a config mapping option in the config-user file

Reply:
@zklaus The idea was indeed to make it possible to specify the path to the new configuration files in config-developer.yml. It is possible to specify the path to config-developer.yml in config-user.yml. I know it is fairly unusual for users to change config-developer.yml, but it is a feature we support. Changing this kind of mapping file is probably similarly unsual, so I think it should be fine to add it there. Most users will not need this as they are not running models anyway.

Reply:
I did have a custom config-developer.yml in the past, but got bitten by updates in the core one too often, so I moved away from that. Which is probably an indication that it would be nice to have a better way to overwrite individual settings. But that's a topic for another day.

So, on topic, I understand that we had that idea to name a mappings or extra facets file in the configuration. But since, we have moved to the idea of supporting multiple mapping files. Should a possible configuration option then point to a directory? Or maybe accept a list of files? Also, should we still support a generic directory in the users configuration directory, or do away with this completely and require all configuration over and above the one shipped in the package to be listed explicitly?

Reply:
> So, on topic, I understand that we had that idea to name a mappings or extra facets file in the configuration

This idea still stands (if `in the configuration` is understood as `in config-dev file`)

> Should a possible configuration option then point to a directory?

I think so

> Or maybe accept a list of files?

I think it doesn't matter 

> Also, should we still support a generic directory in the users configuration directory, or do away with this completely and require all configuration over and above the one shipped in the package to be listed explicitly?

I agree with the latter

Reply:
After giving it some more thought, here is what I did: I added a new configuration option `extra_facets_dir`, but to `config_user.yml`, not `config_developer.yml`. This can take a single path or a list of paths that is appended to the default paths, which are `esmvaltool._config/extra_facets` and `~/.esmvaltool/extra_facets`. Since later entries effectively overwrite earlier ones, like this the user has the option to choose at runtime which replacements should take place, while still giving us all the flexibility to deliver more static versions.

What do you think about it?

Reply:
I'm not entirely convinced yet: to be able to use the extra facets for finding data, the facets from these files have to match the facets used in the DRS which is still only in config-developer.yml. Therefore it seems more logical to keep this together. Of course, it's a bit hard to judge for me, because I don't know how people will use the new feature (yet). Also, it's much harder to change or remove a setting in config-user.yml once it's in, because many more people are affected by it.

Reply:
- The *contents* of the extra_facets files will correspond with the DRS, but their location is quite independent of it. Indeed, it is certainly possible, perhaps even sometimes useful, to have extra_facets that don't appear in the DRS at all and are only used in fixes.
- I don't think many people will be affected by this setting in `config-user.yml`. We could even leave it out of the shipped example entirely and only document it in the documentation since at the moment it is really targeted at a rather specific subset of users.
- In any case, the counterargument is that if you place it in the `config-developer.yml` you increase the number of custom `config-developer.yml` files that the user must maintain from 0 to 2, while the number of custom `config-user.yml` files remains at 2. This creates a certain maintenance problem and in my personal experience this is rather annoying, since then I am forced to update my config-developer files when something is changed in the released version of the file, but it is not trivial to notice when such a change happens.

Reply:
> I don't think many people will be affected by this setting in config-user.yml. We could even leave it out of the shipped example entirely and only document it in the documentation since at the moment it is really targeted at a rather specific subset of users.

Let's do that as a compromise then, to not burden starting users with too many settings.

Reply:
There is one little hiccup: The experimental new config system uses this as the defaults, unlike our production config system which has the defaults hard-coded, and there is a test for the experimental system that trips if a setting is used that doesn't have a default.

Reply:
I removed the setting and fixed the test.